src/views.js

```diff
@@ -381,7 +389,28 @@ function getPreviewForFile(path, thumbnailFactory) {
             thumbnail = thumbnailFactory.generate_thumbnail(uri, mimeType);
 
             if (thumbnail) {
-                thumbnailFactory.save_thumbnail(thumbnail, uri, mtime);
```

Comment:
The comment above that code would need to be adjusted:

// A thumbnail does not currently exist. Ask libgnome-desktop to                                                                           
// create one (currently we do so synchronously) and then                                                                                  
// save the result.